src/services/edusharing/services/EduSharingConnector.js

```diff
@@ -297,15 +313,19 @@ class EduSharingConnector {
 						node.preview.url = await this.getImage(`${node.preview.url}&crop=true&maxWidth=300&maxHeight=300`);
 					}
 
-					// workaround for Edu-Sharing bug, where arrays are as strings "['a,b,c']"
+					// workaround for Edu-Sharing bug, where keywords are like "['a,b,c', 'd \n']"
 					if (
 						node.properties &&
 						node.properties['cclom:general_keyword'] &&
 						node.properties['cclom:general_keyword'][0]
 					) {
-						node.properties['cclom:general_keyword'] = node.properties['cclom:general_keyword'][0]
-							.slice(1, -1)
-							.split(',');
+						const keywords = [];
+						for (const dirtyKeyword of node.properties['cclom:general_keyword']) {
```

Comment:
Please move out this logic to a additional method with naming what it should be do.